### PR TITLE
[FE] Bulk change Alert status

### LINKED ISSUE #1879: [Frontend] Bulk change alert statuses
### Description

Bulk alert status changes need to be implemented in a few areas in the frontend. The backend changes are already in and the API is already in place and used in the latest master branch.

### Related Services

- [x] Alerts API (already merged)

### Designs

[Here](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/87A29D67-F5C4-408F-87E8-DC77FE7F5454?collectionId=0e79571a-428a-4c7f-bb8d-f6c4e457eb57&collectionLayerId=50e28871-e02c-48b4-8775-0f126b5bf15f) and [here](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/AA87C9D5-DFA0-420D-8240-521AB11A7C8B?collectionId=0e79571a-428a-4c7f-bb8d-f6c4e457eb57&collectionLayerId=7fefe367-b34a-4370-9257-357cbdba04dd).

### Acceptance Criteria

A user _must_ have a clear visual indicator to be able to update multiple alert statuses via checkboxes. This can be accomplished by following the designs linked above.

Although rare, the backend can return an error that can be masked in a snackbar triggered in the `onError` handler which displays a friendly error message. Inside the browser's network-console would be a more detailed error message for debugging as we currently have for most async calls.

This design shall be implemented in two places:
-  The main "list alerts" page
-  The "rule details" page
- [Optional] All other areas where we display a list of alerts such as the overview page. Not a requirement as most users don't modify any content on these kinds of pages.

For example:

- User visits the list alerts page and filters for all `INFO` rules
- User clicks the top-level checkbox above the table which selects all (25) items rendered on the page
- User could scroll down, triggering the infinite scroll to poll for more alerts (77 more show up for a total of 102 on the page)
  - If the top-level checkbox was checked, these newly rendered alerts should automatically be checked so the user doesn't have to re-check them.
- User then selects a status level from the dropdown
- User finally clicks the `Apply` button which will fire the request, and if successful, update the items returned in the response payload, and uncheck the boxes. If it failed, show snackbar and leave the alerts checked. In either case, this should **not** trigger a new re-fetch of alerts because we don't want the alerts to disappear from the view as the status update could have been made by mistake.
- The top-level checkbox when checked ON should immediately override any previously selected state for all alert checkboxes and set them to the ON state
- The top-level checkbox when checked OFF should immediately override any previously selected state for all alert checkboxes and set them to the OFF state
- Selecting either the top-level checkbox **or** at least one checkbox next to an alert should change the top-level card to render the number of items selected, the dropdown status field, and the apply button.
  - To exit this context and show back the filter view, a user can refresh the page or uncheck the top-level box.
- The top-level will initially empty, when checked it should show a `-` symbol so the user knows it will deselect all items if clicked again. The other checkboxes will either be empty (blank) when in OFF or show a check symbol when in the ON state.

### PR DESCRIPTION
**Pull request by alexmylonas**
## Description

This is a feature branch that includes PRs #1956 & #2008 and implements the functionality for mass selection and status update for Alerts in **List Alerts** & **Rule Details** page. 

Closes #1879 
### Design

Find mocks [here](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/collections/0e79571a-428a-4c7f-bb8d-f6c4e457eb57)

### Alerts Page
Screenshot 2020-11-13 at 1 10 31 PM
### Single-select
Screenshot 2020-11-13 at 1 10 40 PM
Screenshot 2020-11-13 at 1 11 04 PM
### Bulk Selection
Screenshot 2020-11-13 at 1 11 10 PM
Screenshot 2020-11-13 at 1 11 21 PM
### Rule Details Page
Screenshot 2020-11-13 at 1 11 43 PM
Screenshot 2020-11-13 at 1 11 49 PM
Screenshot 2020-11-13 at 1 11 55 PM

### Checklist:

- [x] Make sure you give a high-level overview of the problem & the solution
- [x] Make sure the related issue is referenced
- [x] Make sure screenshots are added if the UI is changed in any way
- [x] Make sure all of the significant logic is covered by tests
- [x] Make sure your changes are accessible (addition of aria-attributes & correct HTML semantics)
- [x] Make sure you reference the related design document if it exists
- [x] Make sure telemetry is added (analytics & errors)
- [x] ~~Make sure the changes are responsive (appearing decently on devices)~~

### CODE REVIEW

**Comment by alexmylonas** [comment]
@jacknagz & @mattthetraveler This is FE part for bulk update, you can check the screenshots but I also invited both of you to a deployed environment so you can test it out! 

Let me know what you think.

File: web/src/pages/ListAlerts/ListAlertSelection/ListAlertSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+              <FastField
+                name="status"
+                as={FormikCombobox}
+                items={statusOptions}
+                itemToString={filterItemToString}
+                label="Status"
+                placeholder="Select statuses"
+              />
+            </Box>
+            <SubmitButton allowPristineSubmission>Apply</SubmitButton>
```
Comment (3nvi) [inline web/src/pages/ListAlerts/ListAlertSelection/ListAlertSelection.tsx:121]: This needs to be purple according to [the designs](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/commits/bbf6fac0569a248e20b6c94936a9f6164eb76305/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/AA87C9D5-DFA0-420D-8240-521AB11A7C8B?collectionId=0e79571a-428a-4c7f-bb8d-f6c4e457eb57&collectionLayerId=7fefe367-b34a-4370-9257-357cbdba04dd)

**Review by 3nvi — COMMENTED** [review]
Great work @alexmylonas ! I'll let @mattthetraveler do a final pass.

File: web/src/pages/ListAlerts/ListAlertSelection/ListAlertSelection.tsx (hunk 1 above)
Comment (alexmylonas) [reply web/src/pages/ListAlerts/ListAlertSelection/ListAlertSelection.tsx]: Nice catch, updated!

**Review by vorillaz — APPROVED** [review]

**Comment by jacknagz** [comment]
LGTM! This is super clean!

One follow-up item we could do would be shift-clicking to select a range of alerts. Thoughts? (Non-blocking)

**Comment by 3nvi** [comment]
> LGTM! This is super clean!
> 
> One follow-up item we could do would be shift-clicking to select a range of alerts. Thoughts? (Non-blocking)

@jacknagz if can you create a ticket about this with:
 - acceptance criteria
- `team:web` label
- a priority label

Then we'll pick it up soon 😃

**Comment by jacknagz** [comment]
> > LGTM! This is super clean!
> > One follow-up item we could do would be shift-clicking to select a range of alerts. Thoughts? (Non-blocking)
> 
> @jacknagz if can you create a ticket about this with:
> 
> * acceptance criteria
> * `team:web` label
> * a priority label
> 
> Then we'll pick it up soon 😃

Done! #2028 Thanks!

**Comment by mattthetraveler** [comment]
Looks great @alexmylonas and @3nvi ! 

Only thing I noticed is that the user can only select 25 alerts upon filtering unless they scroll down and load more alerts. May be an issue that's out of scope for this; the function works great.

**Comment by 3nvi** [comment]
> Looks great @alexmylonas and @3nvi !
> 
> Only thing I noticed is that the user can only select 25 alerts upon filtering unless they scroll down and load more alerts. May be an issue that's out of scope for this; the function works great.

@mattthetraveler This is unfortunately intentional. 

Right now we don't separate "Select all in view" vs "Select all in system", since we don't have a way to inform the backend to select **every alert**, due to the fact that the backend expects the UI to send a list of specific IDs. Those IDs are not known by the UI unless it has loaded them.

Thus right now 'select all' does 'select all that have been loaded in the UI already'. If we had a way of informing the backend to select all alerts, then we would be able to bypass this limitation. We should have the BE accept a "fixed" string (like `ALL`) or a particular input field (like `selectAll: true`) to instruct it to actually modify all of the alerts (and not the ones present in the UI view)

With this present, we could then bypass certain issues that are present in thee UI and allow thee UI to actually select all alerts (including the ones that come in as you scroll).

Would we want this for 1.15? What's its priority?

**Comment by mattthetraveler** [comment]
Got it, thank you @3nvi . I will put this in our backlog, not a priority for 1.15.

**Pull request merged**